src/www/app/util.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { OperationTimedOut } from '../model/errors';
+import { timeoutPromise } from './util';
+
+const PROMISE_RESOLVED = 1;
+
+describe('timeoutPromise', () => {
+  it('Executes successful promise', () => {
+    timeoutPromise(Promise.resolve(PROMISE_RESOLVED), 100, 'Test Promise')
+        .catch((err) => {
+          fail(`Successful promise was timed out when it should have resolved`);
+        })
+        .then((value) => {
+          expect(value).toEqual(PROMISE_RESOLVED);
+        });
+    const promiseWithTime = new Promise((resolve, _) => {
```

Comment:
This could be a separate test, something like "Executes promise within timeout".